docs/docs/kb/12-plugins.md

```diff
@@ -5,169 +5,250 @@ description: Using and Developing Plugins
+## Developing Plugins
 
-Plugin `Hooks` allow existing ignite commands to be extended with new functionality. Hooks are useful when you want to streamline functionality without needing to run custom scripts after or before a command has been run. this can streamline processes that where once error prone or forgotten all together.
+It's easy to create a plugin and use it immediately in your project. First 
+choose a directory outside your project and run :
 
-### Hook Lifecyle Events
+```sh
+$ ignite plugin scaffold my-plugin
+```
 
-The following are hooks defined which will run on a registered `ignite` commands
+This will create a new directory `my-plugin` that contains the plugin's code,
+and will output some instructions about how to declare your plugin in your
+project. Indeed it's possible to declare a local directory in your project's
+`config.yml`, which has several benefits:
+
+- you don't need to use a git repository during the development of your plugin.
+- the plugin is recompiled each time you run the `ignite` binary in your
+project, if the source files are older than the plugin binary.
+
+Thus, the plugin development workflow is as simple as :
+
+1. scaffold a plugin
+2. declare it in the `config.yml` of a chain (which can be a fresh new chain
+created via `ignite scaffold chain my-chain`)
+3. update plugin code
+4. run `ignite my-command` binary in your chain to compile and run the plugin,
+where `my-command` is a command added by your plugin, or an existing `ignite`
+command with hooks added by your plugin.
+5. go back to 3.
+
+Once your plugin is ready, you can publish it to a git repository, and the
+community can use it by declaring this git repository path in their chain's
+`config.yml`.
+
+Now let's detail how to update your plugin's code.
+
+### The plugin interface
+
+The `ignite` plugin system uses `github.com/hashicorp/go-plugin` under the hood,
+which implies to implement a predefined interface:
+
+```go title=ignite/services/plugin/interface.go
+// An ignite plugin must implements the Plugin interface.
+type Interface interface {
+	// Manifest declares the plugin's Command(s) and Hook(s).
+	Manifest() (Manifest, error)
+
+	// Execute will be invoked by ignite when a plugin Command is executed.
+	// It is global for all commands declared in Manifest, if you have declared
+	// multiple commands, use cmd.Path to distinguish them.
+	Execute(cmd ExecutedCommand) error
+
+	// ExecuteHookPre is invoked by ignite when a command specified by the Hook
+	// path is invoked.
+	// It is global for all hooks declared in Manifest, if you have declared
+	// multiple hooks, use hook.Name to distinguish them.
+	ExecuteHookPre(hook ExecutedHook) error
+
+	// ExecuteHookPost is invoked by ignite when a command specified by the hook
+	// path is invoked.
+	// It is global for all hooks declared in Manifest, if you have declared
+	// multiple hooks, use hook.Name to distinguish them.
+	ExecuteHookPost(hook ExecutedHook) error
```

Comment:
were arguments removed from hooks? if so the examples below will need to be updated as well as they still contain an `args` param

Reply:
good catch I missed that [018db4e](https://github.com/ignite/cli/pull/3066/commits/018db4eaa201c003ff867557042ec68a039b6e22)